### PR TITLE
feat(policies): Remove individual policies from the sidebar 

### LINKED ISSUE #462: Review policy sidebar item
### Description

The sidebar is very loaded with a large policy list (some like MeshGateway and MeshGatewayRoutes are not even policies).

Could we review this to be a page instead? What is the rest like?

### PR DESCRIPTION
**Pull request by johncowen**
Resolves #462

Removes all the Policies from the sidebar, and replaces it with one `Policies` link which redirects to the first Policy list with non-zero items.

Once on the policy listing, switching policies is achieved using a filter like menu which shows you only the policies of a particular type. We were too-ing and fro-ing about whether to only show policy types that have actual items in them. I've left it showing all policy items in this PR. So we end up with this kind of thing (minus the policies in the sidebar)

policy-selection

Other additions:

1. RouterLink stubbing to make App.spec pass with the new link in it (see convo offline)
2. Moved PolicyType sorting to the Vuex store as soon as they are set. This way, as long as we access these through the store, they will always be consistently sorted. I don't think this has any consequence elsewhere.

Future work:

We've talked about moving the URL structure here to use dynamic segments instead of looping through all types and adding a separate route for each, additionally name spacing all policy type routes under a `policies` segment, i.e. `/mesh/default/policies/:type`. You'll see the redirect is kinda of starting this off but the replacement of the looping with a dynamic segment will be follow up work.

Signed-off-by: John Cowen <john.cowen@konghq.com>

### CODE REVIEW

**Comment by johncowen** [comment]
I've added the things we chatted about yesterday:

- Scrollable, typeahead menu
- Fading out the policy types that have no items
- Added the total count to the Policy item in the side bar

Also updated the screencap in the description above.

Gonna review the comments already on the PR now also

File: src/app/policies/views/PolicyView.vue [hunk 1]
```diff
@@ -189,6 +207,18 @@ const tableData = ref<{ headers: TableHeader[], data: any[] }>({
 })
 
 const policy = computed(() => store.state.policiesByPath[props.policyPath])
+const policies = computed(() => {
+  return store.state.policies.map((item) => {
+    return {
+      /* length: store.state.sidebar.insights.mesh.policies[item.name], */
```
[the remark this answers is not in the document]
Comment (johncowen) [reply src/app/policies/views/PolicyView.vue]: Thanks! I figured I'd need this, and the new updates do need this, thanks for the nudge tho!

File: src/router/router.ts [hunk 2]
```diff
@@ -176,6 +186,21 @@ export function createRouter(baseGuiPath: string = '/', policyDefinitions: Polic
...
+          redirect: (to: RouteLocation): RouteLocationRaw => {
+            let item = store.state.policies
+              .find((item) => store.state.sidebar.insights.mesh.policies[item.name] !== 0)
+            if (item === undefined) {
+              item = store.state.policies[0]
+            }
+            return {
+              name: item.path,
+              params: to.params,
+            }
```
[the remark this answers is not in the document]
Comment (johncowen) [reply src/router/router.ts:201]: No prob. I can't think of an occasion where query params would get set on a redirect though? You can't interact with the page at this route so query params would never get added. I'm gonna apply the suggestion anyway though as I don't think it matters either way.

[the remark this answers is not in the document]
Comment (kleinfreund) [reply src/router/router.ts]: No, right, this doesn’t fix anything now or possibly in the future, but I think it’s not a bad idea as a general defensive rule for dealing with routes programmatically. In the end pretty arbitrary though, yes.

File: src/router/router.ts [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
-    }),
-    component: () => import('@/app/policies/views/PolicyView.vue'),
-  }))
+  return policies.map((policy) => {
+    return {
+      path: policy.path,
+      name: policy.path,
+      meta: {
+        shouldReRender: true,
+        title: policy.pluralDisplayName,
```
Comment (kleinfreund) [inline src/router/router.ts:23]: **Issue**: Currently, the sidebar item is not highlighted which can be fixed by marking the `parent` in its meta properties. We use this field for things like this and as part of the breadcrums.

```suggestion
        title: policy.pluralDisplayName,
        parent: 'policies',
```

File: src/router/router.ts [hunk 4]
```diff
@@ -176,6 +186,21 @@ export function createRouter(baseGuiPath: string = '/', policyDefinitions: Polic
             },
           ],
         },
+        {
+          path: 'policies',
+          name: 'policies',
```
Comment (kleinfreund) [inline src/router/router.ts:191]: **Suggestion**: Adds the title needed to have this show up in the breadcrumbs correctly (related to the suggestion with the `parent` field).

```suggestion
          name: 'policies',
          meta: {
            title: 'Policies',
          },
```

**Review by kleinfreund — CHANGES_REQUESTED** [review]
Found a minor issue, but the rest looks good to go.

**Comment by johncowen** [comment]
Oh good catch, let me check that here so I can see, 2 tics

**Comment by johncowen** [comment]
K those 2 are done!

**Review by kleinfreund — APPROVED** [review]

**Pull request merged**